### PR TITLE
[DBC-210] Integrating tkey-bridge with WebView

### PR DESCRIPTION
**Pull request by quocnd**
## Description
- Implement tkey-bridge with WebView
- Login successfully

## Note
- Store `DeviceShare` `PostboxKey` in `SecuredSharedPreferences` will be replaced by `KeyStore`
- Currently using a timeout 60 seconds on Android. Will remove it once javascript handles error or timeout
- Will move `core` package into `wallet-core`

## Screenshots

https://user-images.githubusercontent.com/10803943/139391799-face5126-1dde-43e0-96a9-596947a76ce4.mp4

### CODE REVIEW

File: versions.gradle [hunk 1]
```diff
@@ -21,7 +21,7 @@ versions.DESUGAR_JDK = '1.1.5'
 versions.FIREBASE = '28.4.1'
 versions.FRAGMENT_KTX = '1.3.6'
 versions.HILT = '2.38.1'
-versions.KOTLIN = '1.5.31'
+versions.KOTLIN = '1.6.0-RC'
```
Comment (cutehackers) [inline versions.gradle:24]: Is this stable at the moment?

File: app/src/main/java/io/mvlchain/tada/wallet/app/ui/authguide/AuthGuideFragment.kt [hunk 2]
```diff
@@ -15,31 +15,123 @@
...
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class AuthGuideFragment : BaseFragment<FragmentAuthGuideBinding>() {
 
     override val viewModel: AuthGuideViewModel by viewModels()
 
+    @Inject
+    lateinit var loginInfo: LoginInfo
+    private lateinit var mvlAuthManager: MvlAuthManager
```
Comment (cutehackers) [inline app/src/main/java/io/mvlchain/tada/wallet/app/ui/authguide/AuthGuideFragment.kt:45]: It's a great implementation for Mvl Auth features, In addition, I reckon that you can improve mvlAuthManager as lifecycle aware class in the future. Not now, I'll be in the future. 

refer to these links. 
https://developer.android.com/topic/libraries/architecture/lifecycle
https://www.raywenderlich.com/22025947-lifecycle-aware-components-using-android-jetpack

File: shared/src/main/java/io/mvlchain/tada/wallet/shared/domain/GetPinCodeUseCase.kt [hunk 3]
```diff
@@ -15,15 +15,15 @@
...
 
-import io.mvlchain.tada.wallet.shared.data.repository.PinCodeRepository
+import io.mvlchain.tada.wallet.shared.data.repository.CredentialRepository
 import io.mvlchain.tada.wallet.shared.di.IoDispatcher
 import io.mvlchain.tada.wallet.shared.lib.Resource
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.flow.Flow
 import javax.inject.Inject
 
 class GetPinCodeUseCase @Inject constructor(
```
Comment (cutehackers) [inline shared/src/main/java/io/mvlchain/tada/wallet/shared/domain/GetPinCodeUseCase.kt:25]: Do we need this useCase? I think what we only need is Verifying pincode and there is no such cases that reuires pincode directly. need you opinion.

**Review by cutehackers — APPROVED** [review]

File: versions.gradle (hunk 1 above)
Comment (quocnd) [reply versions.gradle:24]: This is not a stable release at this time, but it fixes issue incompatible with JDK16+
https://youtrack.jetbrains.com/issue/KT-45545
I will let an eye on this and will upgrade once it has a stable version.

File: app/src/main/java/io/mvlchain/tada/wallet/app/ui/authguide/AuthGuideFragment.kt (hunk 2 above)
Comment (quocnd) [reply app/src/main/java/io/mvlchain/tada/wallet/app/ui/authguide/AuthGuideFragment.kt:45]: Noted, thanks

File: shared/src/main/java/io/mvlchain/tada/wallet/shared/domain/GetPinCodeUseCase.kt (hunk 3 above)
Comment (quocnd) [reply shared/src/main/java/io/mvlchain/tada/wallet/shared/domain/GetPinCodeUseCase.kt:25]: You're right, for this PR we don't need Pincode, just need a Boolean value. 
But we need it in PasswordVerifierFragment that requires `storedPinCode` to compare. So I think we can keep it

**Pull request merged**